### PR TITLE
[WIP] Add apply diff-last-applied subcommand

### PR DESCRIPTION
**Pull request by shiywang**
https://github.com/kubernetes/community/pull/287
currently `apply diff-last-applied` will invoke `git diff --no-index`, if there's no `git`, it will degenerate to use `diff` also you can use a customize diff tool by setting env : `KUBECTL_EXTERNAL_DIFF`
```console
[root@shiywang template]# ./kubectl apply diff-last-applied -f nginx.yaml 
diff --git a/tmp/kubectl-diff-yd44r.yaml b/tmp/kubectl-diff-0wox0.yaml
index 46aa188..d44cab2 100644
--- a/tmp/kubectl-diff-yd44r.yaml
+++ b/tmp/kubectl-diff-0wox0.yaml
@@ -8,7 +8,7 @@ spec:
   template:
     metadata:
       labels:
-        app: nginx123
+        app: nginx
     spec:
       containers:
       - image: gcr.io/google_containers/nginx-slim:0.8
[root@shiywang template]# 
```
@kubernetes/sig-cli-pr-reviews 
@mengqiy  could you have an early review on this ? thanks

### CODE REVIEW

**Comment by mengqiy** [comment]
@shiywang Please rebase and I will start to review. Thanks.

**Comment by mengqiy** [comment]
@shiywang is this one ready to review as well?

File: pkg/kubectl/cmd/apply_diff_last_applied.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	Output           string
+	Codec            runtime.Encoder
+	Factory          cmdutil.Factory
+	Out              io.Writer
+	ErrOut           io.Writer
+	ext              string
+}
+
+var (
+	applyDiffLastAppliedLong = templates.LongDesc(`
```
Comment (pwittrock) [inline pkg/kubectl/cmd/apply_diff_last_applied.go]: I think we need a 3-way diff - config vs new config vs cluster live

**Comment by pwittrock** [comment]
@shiywang Is this still something you are working on?

**Comment by shiywang** [comment]
@mengqiy @pwittrock yes, I'm still working on this, will rebase after apply edit get merged

**Comment by pwittrock** [comment]
@shiywang Ok thanks.  Quick reminder that feature freeze is 2 weeks from tomorrow.

**Comment by shiywang** [comment]
close this due to rebase is aching, I will submit a new pr

**Pull request closed without merging**